conformance/ion_encoding/symtab.ion

```diff
@@ -0,0 +1,32 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Basic local symbols inside $ion_encoding
+
+(ion_1_1
+  (each (toplevel $ion_encoding::((symbol_table ["a", "b"])))
+        (toplevel $ion_encoding::((symbol_table ["a"] ["b"])))
+        (toplevel $ion_encoding::((symbol_table ["a"] [] ["b"])))
+        (toplevel $ion_encoding::((symbol_table ["a"] null.list ["b"])))
```

Comment:
Do we want to support `null` and non-list values (by ignoring them)?

Reply:
In general I think it's best to be strict/conservative/typesafe in the new directives, because any "ignore" rule will tend to have a bunch of exceptions.  Symbols and S-expressions generally denote language features (variable refs and keyword-clauses), some contexts will care about some other types, etc.

In general it's safest long-term to be consistently strict in all cases, than to "ignore" consistently everywhere when the type is unexpected.  The latter makes it harder to extend with new functionality in the future without changing the meaning of existing constructs, and at this stage I really want to protect our ability to evolve things.

Reply:
Fully agreed. I was really confused by this because I had missed that this test case was expected to `signal`. 😅

Reply:
Sorry, I don't understand what you mean.  Can you give an example?

Reply:
To be clear, what this test shows is that `(symbol_table null.list)` behaves like `(symbol_table [])`.
In this specific context that's equivalent to "is ignored".

I'll add cases showing that types other than symbol and list inside `symbol_table` is erroneous.